src/Microsoft.ML.CodeGenerator/Utils.cs

```diff
@@ -309,12 +284,51 @@ internal static IList<string> GenerateClassLabels(ColumnInferenceResults columnI
                     result.Add($"[ColumnName(\"{columnName}\"), LoadColumn({column.Source[0].Min})]");
                 }
                 sb.Append(" ");
-                sb.Append(Utils.Normalize(column.Name));
-                sb.Append("{get; set;}");
+                string normalizedColumnName = Utils.Normalize(column.Name);
+                // Put placeholder for normalized and unique version of column name
+                if (!duplicateColumnNamesExist && normalizedColumnNames.Contains(normalizedColumnName))
+                    duplicateColumnNamesExist = true;
+                normalizedColumnNames.Add(normalizedColumnName);
                 result.Add(sb.ToString());
                 result.Add("\r\n");
             }
+            for (int i = 1; i < result.Count; i+=3)
```

Comment:
Could you make sure the generated label are identical for other source files in generated project?
For example, here in Program.tt
https://github.com/mstfbl/machinelearning/blob/42dfa105fddf04ab23d0e574580edf6f99a955a6/src/Microsoft.ML.CodeGenerator/Templates/Console/PredictProgram.tt#L47

and here: (SampleData also call into Util.Normalize)
https://github.com/mstfbl/machinelearning/blob/42dfa105fddf04ab23d0e574580edf6f99a955a6/src/Microsoft.ML.CodeGenerator/Templates/Console/PredictProgram.tt#L27

Reply:
Currently, these generated labels are generated using `Utils.Normalize` in [GenerateSampleData](https://github.com/dotnet/machinelearning/blob/1296c29c9e08cbe0720b021144d2d08fa2c04a8e/src/Microsoft.ML.CodeGenerator/Utils.cs#L50-L69), and hence are not added the `_col_x` suffix for conflicting column names. I am making an issue to track this change. Thanks!